modules/fedimint-ln-client/src/db.rs

```diff
@@ -247,18 +269,67 @@ pub(crate) fn get_v3_migrated_state(
                 out_points: v0.out_points,
                 error_reason: "unknown error (database migration)".to_string(),
             };
-            let new_pay = LightningPayStateMachine {
+            let new_pay = LightningPayStateMachineOld {
                 common,
                 state: LightningPayStates::Refund(v1),
             };
-            let new_sm = LightningClientStateMachines::LightningPay(new_pay);
-            let bytes = new_sm.consensus_encode_to_vec();
-            Ok(Some((bytes, operation_id)))
+            let pay_bytes = new_pay.consensus_encode_to_vec();
+            let mut sm_bytes: Vec<u8> = Vec::new();
+            1_u16.consensus_encode(&mut sm_bytes).unwrap();
+            let len = u16::try_from(pay_bytes.len()).unwrap();
+            len.consensus_encode(&mut sm_bytes).unwrap();
+            sm_bytes.extend_from_slice(&pay_bytes);
+            Ok(Some((sm_bytes, operation_id)))
         }
         _ => Ok(None),
     }
 }
 
+/// Adds the `amount` field to `LightningPayCommon`
+pub(crate) fn get_v4_migrated_state(
```

Comment:
I know its cumbersome but would be great to have a test for this, for extra confidence